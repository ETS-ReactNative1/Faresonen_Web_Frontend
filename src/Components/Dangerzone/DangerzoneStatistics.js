import React from "react";
import PropTypes from "prop-types";
import { Statistic } from "semantic-ui-react";

const socketConnected = (connected = false) => {
  if (connected) {
    return (
      <Statistic>
        <Statistic.Value>
          <i className="green server icon" />
        </Statistic.Value>
        <Statistic.Label>Tilkoblet server</Statistic.Label>
      </Statistic>
    );
  }
  return (
    <Statistic>
      <Statistic.Value>
        <i className="red server icon" />
      </Statistic.Value>
      <Statistic.Label>Frakoblet server</Statistic.Label>
    </Statistic>
  );
};

const DangerzoneStatistics = props => (
  <Statistic.Group widths="three">
    {socketConnected(props.socketConnected)}
    <Statistic>
      <Statistic.Value>{props.eliteserien}</Statistic.Value>
      <Statistic.Label>Eliteserie-spillere</Statistic.Label>
    </Statistic>
    <Statistic>
      <Statistic.Value>{props.obosligaen}</Statistic.Value>
      <Statistic.Label>OBOSliga-spillere</Statistic.Label>
    </Statistic>
  </Statistic.Group>
);

DangerzoneStatistics.propTypes = {
  socketConnected: PropTypes.bool.isRequired,
  eliteserien: PropTypes.number.isRequired,
  obosligaen: PropTypes.number.isRequired
};
export default DangerzoneStatistics;
