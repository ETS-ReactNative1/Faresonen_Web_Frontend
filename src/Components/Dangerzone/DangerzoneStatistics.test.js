import React from "react";
import renderer from "react-test-renderer";
import DangerzoneStatistics from "./DangerzoneStatistics";

describe("<DangerzoneStatistics />", () => {
  it("Should render properly", () => {
    const tree = renderer.create(
      <DangerzoneStatistics eliteserien={32} obosligaen={0} socketConnected />,
    );
    expect(tree).toMatchSnapshot();
  });
});
