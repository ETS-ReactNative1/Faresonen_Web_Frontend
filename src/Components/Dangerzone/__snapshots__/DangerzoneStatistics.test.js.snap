// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DangerzoneStatistics /> Should render properly 1`] = `
<div
  className="ui three statistics"
>
  <div
    className="ui statistic"
  >
    <div
      className="value"
    >
      <i
        className="green server icon"
      />
    </div>
    <div
      className="label"
    >
      Tilkoblet server
    </div>
  </div>
  <div
    className="ui statistic"
  >
    <div
      className="value"
    >
      32
    </div>
    <div
      className="label"
    >
      Eliteserie-spillere
    </div>
  </div>
  <div
    className="ui statistic"
  >
    <div
      className="value"
    >
      0
    </div>
    <div
      className="label"
    >
      OBOSliga-spillere
    </div>
  </div>
</div>
`;
